filter.ts

```diff
@@ -15,7 +15,11 @@ export function matchFilter(
   filter: Filter,
   event: Event & {id: string}
 ): boolean {
-  if (filter.ids && filter.ids.indexOf(event.id) === -1) return false
+  if (filter.ids && filter.ids.indexOf(event.id) === -1) {
+    if (!filter.ids.some(prefix => prefix.length < 64 && event.id.startsWith(prefix))) {
```

Comment:
remove this `prefix.length` check, it's unnecessary

Reply:
I'll remove, I just wanted to make it explicit that prefix is only if it's smaller than 64.

> The ids and authors lists contain lowercase hexadecimal strings, which may either be an exact 64-character match, or a prefix of the event value.

https://github.com/nostr-protocol/nips/blob/master/01.md

Reply:
removed